includes/wc-core-functions.php

```
@@ -233,12 +233,20 @@ function wc_get_template( $template_name, $args = array(), $template_path = '',
 	);
 
 	if ( ! empty( $args ) && is_array( $args ) ) {
+		if ( isset( $args['action_args'] ) ) {
+			wc_doing_it_wrong(
+				__FUNCTION__,
+				__( 'action_args should not be overwritten when calling wc_get_template.', 'woocommerce' ),
+				'3.6.0'
+			);
+			return;
```

Why not soft-deprecate it? Just throw the warning and unset it?